src/proteus/dswx_hls.py

```diff
@@ -4156,6 +4283,53 @@ class in the browse image. False to display PSW-Agg.
                                 shoreline_shapefile, geotransform,
                                 projection, length, width)
 
+    # print input HLS product spatial and cloud coverage
+    logger.info(f'data coverage:')
+    if 'INPUT_HLS_PRODUCT_SPATIAL_COVERAGE' in dswx_metadata_dict.keys():
+        hls_product_spatial_coverage = dswx_metadata_dict[
+            'INPUT_HLS_PRODUCT_SPATIAL_COVERAGE']
+        logger.info(f'    input HLS product spatial coverage [%]:'
+                    f' {hls_product_spatial_coverage}')
+
+    if 'INPUT_HLS_PRODUCT_CLOUD_COVERAGE' in dswx_metadata_dict.keys():
+        hls_product_cloud_coverage = dswx_metadata_dict[
+            'INPUT_HLS_PRODUCT_CLOUD_COVERAGE']
+        logger.info(f'    input HLS product cloud coverage [%]:'
+                    f' {hls_product_cloud_coverage}')
+
+    # compute a preliminary cloud layer, i.e., without snow/ice class
+    preliminary_cloud_layer = _compute_preliminary_cloud_layer(
+        fmask, mask_adjacent_to_cloud_mode)
+
+    # create ocean mask
+    if shoreline_shapefile is not None:
+        ocean_mask = _create_ocean_mask(shoreline_shapefile,
+                                        ocean_masking_shoreline_distance_km,
+                                        scratch_dir, geotransform, projection,
+                                        length, width,
+                                        temp_files_list=temp_files_list)
+
+        # update valid_array
+        valid_array = np.logical_and(valid_array, ocean_mask)
+
+    # create variables to compute spatial and cloud coverage
+    n_valid = np.sum(valid_array)
+    n_cloud_and_valid = np.sum(((preliminary_cloud_layer) != 0) & (valid_array))
+    del valid_array
+    total_number_of_pixels = length * width
+
+    # use floor integer to compute spatial and cloud coverage
+    spatial_coverage = int(100 * float(n_valid) / total_number_of_pixels)
+    cloud_coverage = int(100 * float(n_cloud_and_valid) / n_valid)
+
+    # print spatial and cloud coverage
+    logger.info(f'    spatial coverage [%]:  {spatial_coverage}')
+    logger.info(f'    cloud coverage [%]:  {cloud_coverage}')
+
+    # update DSWx-HLS metadata dictionary
+    dswx_metadata_dict['SPATIAL_COVERAGE'] = spatial_coverage
+    dswx_metadata_dict['CLOUD_COVERAGE'] = cloud_coverage
+
```

Comment:
Git is not allowing me to comment on the relevant lines, but could you please confirm that it is correct that there is no fill data value for the SHAD.tif layer?

The Product Spec is not specific about this, and running `gdalinfo` on one of the *SHAD.tif output files shows that there is no fill data value specified, and then in the color table 255 is fully visible (not transparent). Thanks!

Reply:
Hi @nemo794 , thank you for checking, but I think the shadow layer doesn't have fill value. We expect the DEM to be available globally. If it's not available for the area, the `_check_ancillary_inputs()` should raise an error.

Reply:
Sounds good! And thank you for updating the Product Spec document to match. 🚀